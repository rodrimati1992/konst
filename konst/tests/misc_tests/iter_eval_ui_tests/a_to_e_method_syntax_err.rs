const R: core::ops::Range<usize> = 0..10;
const S: &[u8] = &[3, 5];

const fn whatever1_return<T>(a0: impl Sized) -> T {
    std::mem::forget(a0);
    unimplemented!()
}


const _: () = {
    konst::iter::eval!(R, any());
    konst::iter::eval!(R, any(||false));
    konst::iter::eval!(R, any(|_|));
    konst::iter::eval!(R, any(|_|(), 100));
    konst::iter::eval!(R, any(|a, b|false));
    konst::iter::eval!(R, any({whatever1_return}));
};

const _: () = {
    konst::iter::eval!(R, all());
    konst::iter::eval!(R, all(||false));
    konst::iter::eval!(R, all(|_|(), 100));
    konst::iter::eval!(R, all(|a, b|false));
    konst::iter::eval!(R, all({whatever1_return}));
};

const _: () = {
    konst::iter::eval!(S, copied(10), next());
};

const _: () = {
    konst::iter::eval!(R, count(10));
};

const _: () = {
    konst::iter::eval!(R, enumerate(10), next());
};


fn main(){}
