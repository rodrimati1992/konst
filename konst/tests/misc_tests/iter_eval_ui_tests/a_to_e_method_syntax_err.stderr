error: `any` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:11:5
   |
11 |     konst::iter::eval!(R, any());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `any` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:12:5
   |
12 |     konst::iter::eval!(R, any(||false));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `any` expects valid closure syntax, passed closure without return value
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:13:5
   |
13 |     konst::iter::eval!(R, any(|_|));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_no_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `any` expects no arguments after closure argument
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:14:5
   |
14 |     konst::iter::eval!(R, any(|_|(), 100));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_trailing_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `any` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:15:5
   |
15 |     konst::iter::eval!(R, any(|a, b|false));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `any` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:16:5
   |
16 |     konst::iter::eval!(R, any({whatever1_return}));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `all` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:20:5
   |
20 |     konst::iter::eval!(R, all());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `all` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:21:5
   |
21 |     konst::iter::eval!(R, all(||false));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `all` expects no arguments after closure argument
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:22:5
   |
22 |     konst::iter::eval!(R, all(|_|(), 100));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_trailing_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `all` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:23:5
   |
23 |     konst::iter::eval!(R, all(|a, b|false));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `all` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:24:5
   |
24 |     konst::iter::eval!(R, all({whatever1_return}));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid argument(s) for `copied` method: `10`
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:28:5
   |
28 |     konst::iter::eval!(S, copied(10), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::iter::__eval2_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid argument(s) for `count` method: `10`
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:32:5
   |
32 |     konst::iter::eval!(R, count(10));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::iter::__eval2_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid argument(s) for `enumerate` method: `10`
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:36:5
   |
36 |     konst::iter::eval!(R, enumerate(10), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::iter::__eval2_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:14:5
   |
14 |     konst::iter::eval!(R, any(|_|(), 100));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `bool`, found `()`
   |
   = note: this error originates in the macro `$crate::__eval_closure` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/iter_eval_ui_tests/a_to_e_method_syntax_err.rs:22:5
   |
22 |     konst::iter::eval!(R, all(|_|(), 100));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `bool`, found `()`
   |
   = note: this error originates in the macro `$crate::__eval_closure` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)
